var/spack/repos/builtin/packages/xsdk-examples/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class XsdkExamples(CMakePackage):
+    """xSDK Examples show usage of libraries in the xSDK package."""
+
+    homepage = 'http://xsdk.info'
+    url      = 'https://github.com/xsdk-project/xsdk-examples/archive/v0.1.0.tar.gz'
+
+    maintainers = ['acfisher', 'balay', 'balos1', 'luszczek']
+
+    version('0.1.0', sha256='d24cab1db7c0872b6474d69e598df9c8e25d254d09c425fb0a6a8d6469b8018f')
+
+    depends_on('xsdk@0.5.0', when='@0.1.0')
+
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            '-DCMAKE_C_COMPILER=%s' % spec['mpi'].mpicc,
+            '-DMPI_DIR=%s' % spec['mpi'].prefix,
+            '-DSUNDIALS_DIR=%s'     % spec['sundials'].prefix,
+            '-DPETSC_DIR=%s'         % spec['petsc'].prefix,
+            '-DPETSC_INCLUDE_DIR=%s' % spec['petsc'].prefix.include,
+            '-DPETSC_LIBRARY_DIR=%s' % spec['petsc'].prefix.lib,
+            '-DSUPERLUDIST_INCLUDE_DIR=%s' %
+            spec['superlu-dist'].prefix.include,
+            '-DSUPERLUDIST_LIBRARY_DIR=%s' % spec['superlu-dist'].prefix.lib,
+        ]
+        if 'trilinos' in spec:
```

Comment:
This isn't supposed to be a variant, right? If it's a dependency, I think what you want is:
```suggestion
        if '^trilinos' in spec:
```
What you already have may work as well, I'm not sure.

Reply:
thanks @adamjstewart.  The current code did work [as in it built this trilinos examples] - however I don't completely understand the cmake mechanism here. @luszczek can confirm.

There is also this ` spec['trilinos'].prefix` - which indicates `if 'trilinos' in spec` is valid?

Reply:
I tried a build with this change  i.e `if '^trilinos' in spec:` - and that worked as-well.
For now - will keep the current code [and merge in] and update it as issues come up.

Reply:
`xsdk-examples` depends on `xsdk` which has `trilinos` variant. I tried against both and `xsdk-examples` works only when I have the conditional. Perhaps adding a `trilinos` variant here would a better option in the future.